### ISSUE TITLE
arrowToBatch() error when you do Decimal(38, 18)

### ISSUE THREAD

**Issue by Yifeng-Sigma** [snowflakedb/gosnowflake#878]
https://github.com/apache/arrow/issues/36998#issuecomment-1664757532

https://github.com/snowflakedb/gosnowflake/blob/master/converter.go#L986C1-L986C1
When receiving an int64 that exceeds the maximum integer that float64 can represent, we will have an error when `compute.Divide()`, the reason are as above. 

To reproduce you can do `select TO_NUMBER('0.1', 38, 17) ` with arrow batch enabled. 

```
else if srcColumnMeta.Scale != 0 && col.DataType().ID() == arrow.INT64 {
 				values := col.(*array.Int64).Int64Values()
 				floatValues := make([]float64, len(values))

 				for i, val := range values {
 					floatValues[i], _ = intToBigFloat(val, srcColumnMeta.Scale).Float64()
 				}
 				builder := array.NewFloat64Builder(memory.NewCheckedAllocator(memory.NewGoAllocator()))
 				builder.AppendValues(floatValues, nil)
 				newCol = builder.NewArray()
 				builder.Release()
 				defer newCol.Release()
 			}
```
fixed this.

**Comment by sfc-gh-dszmolka** [comment]
hi and thank you for raising this Issue with us - also really appreciate linking the Apache one and sending the reproduction and proposed solution ! We'll take a look and keep this Issue updated.

**Comment by sfc-gh-dszmolka** [comment]
fix released with 2024 March release, version 1.9.0. thank you for your contribution!
